### DISCUSSION TITLE
SOCKS5 Support in anacrolix/torrent

### DISCUSSION THREAD

**Discussion by varbhat** [anacrolix/torrent#557] [Q&A]
Please see https://github.com/varbhat/exatorrent/discussions/35 .

So, does anacrolix/torrent support SOCKS5 ? I think that http.ProxyFromEnvironment already supports SOCKS5 (see link that i have mentioned above).

I have grepped for all instances where ClientConfig's HTTPProxy and found that it's respected for trackers and few other things (what about UDP,TCP connections? do they support?).

@anacrolix

**Comment by anacrolix** [comment]
Thanks, I'm actively interested in this now. I've reopened https://github.com/anacrolix/torrent/issues/160 as the implicit dial support implemented in the PR there has changed.

**Comment by anacrolix** [comment]
HTTPProxy is currently only used by trackers and a few other things, as you described.
